modules/theme-tools/content-options/blog-display.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * The functions to display Content or Excerpt in a theme.
+ */
+
+/**
+ * If the theme doesn't support 'jetpack-content-options', don't continue.
+ */
+if ( ! current_theme_supports( 'jetpack-content-options' ) ) {
+	return;
+}
+
+/**
+ * Get the Blog Display setting.
+ * If theme is using both 'Content' and 'Excerpt' then this setting will be called 'Mixed'.
+ */
+$options      = get_theme_support( 'jetpack-content-options' );
+$blog_display = ( ! empty( $options[0]['blog-display'] ) ) ? $options[0]['blog-display'] : null;
+$blog_display = preg_grep( '/^(content|excerpt)$/', (array) $blog_display );
+sort( $blog_display );
+$blog_display = implode( ', ', $blog_display );
+$blog_display = ( 'content, excerpt' === $blog_display ) ? 'mixed' : $blog_display;
+
+/**
+ * If the theme doesn't support 'jetpack-content-options[ 'blog-display' ]', don't continue.
+ */
+if ( ! in_array( $blog_display, array( 'content', 'excerpt', 'mixed' ) ) ) {
+	return;
+}
+
+/**
+ * Apply Content filters.
+ */
+function jetpack_blog_display_custom_excerpt( $content ) {
+	$post = get_post();
+	if ( empty( $post->post_excerpt ) ) {
+		$text = strip_shortcodes( $post->post_content );
+		$text = str_replace( ']]>', ']]&gt;', $text );
+		$text = strip_tags( $text );
+		$excerpt_length = apply_filters( 'excerpt_length', 55 );
```

Comment:
Could you add a docblock for that filter? We use the following format for filters from Core:

`/** This filter is documented in wp-includes/relevant/file/here.php */`